Sources/PayPal/PayPalClient.swift

```diff
@@ -0,0 +1,79 @@
+import UIKit
+
+@_implementationOnly import PayPalCheckout
+
+#if canImport(PaymentsCore)
+import PaymentsCore
+#endif
+
+/// PayPal Paysheet to handle PayPal transaction
+public class PayPalClient {
+
+    private let config: CoreConfig
+    private let returnURL: String
+
+    // swiftlint:disable identifier_name
+    private let CheckoutFlow: PayPalUIFlow.Type
+    // swiftlint:enable identifier_name
+
+    // TODO: I don't think we can override a lazy var in our tests ... or can we?
+    lazy var nativeCheckoutSDKConfig: CheckoutConfig = {
+        CheckoutConfig(
+            clientID: config.clientID,
+            returnUrl: returnURL,
+            environment: config.environment.toNativeCheckoutSDKEnvironment()
+        )
+    }()
+
+    /// Initialize a PayPalClient to process PayPal transaction
+    /// - Parameters:
+    ///   - config: The CoreConfig object
+    ///   - returnURL: The return URL provided to the PayPal Native UI experience. Used as part of the authentication process to identify your application. This value should match the one set in the `Return URLs` section of your application's dashboard on your [PayPal developer account](https://developer.paypal.com)
+    public convenience init(config: CoreConfig, returnURL: String) {
+        self.init(config: config, returnURL: returnURL, checkoutFlow: Checkout.self)
+    }
+
+    init(config: CoreConfig, returnURL: String, checkoutFlow: PayPalUIFlow.Type) {
```

Comment:
In the `APIClient.swift` we have the internal vs public initializers structured a little differently.  They both achieve the same thing, but we should stick to one pattern and keep it consistent throughout the repo.

Reply:
yeah removed the `convenience` init. May be worth revisiting in the future though to refactor that pattern to use the convenience inits though - to keep our initialization logic in a single place and avoid the chance that we leave something out of one of our initializers, since all of the public facing convenience inits would call one of our required (and tested) inits under the hood.